olive/passes/pytorch/slicegpt.py

```diff
@@ -128,16 +129,20 @@ def _run_for_config(
             100 * (1 - new_embedding_dim / model_adapter.hidden_size),
         )
 
-        train_dataset = get_dataset(config.calibration_data_config.name)["train"]
-        train_loader = prepare_dataloader(
-            dataset=train_dataset,
-            tokenizer=tokenizer,
-            max_seqlen=config.calibration_max_seqlen,
-            batch_size=config.calibration_batch_size,
-            nsamples=config.calibration_nsamples,
-            varied_seqlen=config.varied_seqlen,
-            seed=config.seed,
-        )
+        data_config = validate_config(config.calibration_data_config, DataConfig)
+        dataloader = data_config.to_data_container().create_dataloader(data_root)
+        dataset = [
+            {
+                "input_ids": data[0]["input_ids"].squeeze(),
+                "attention_mask": data[0]["attention_mask"].squeeze(),
+                "labels": data[0]["input_ids"].clone().squeeze(),
```

Comment:
not an issue here since all of the data if of max length so there is no padding, etc. But the data already has "labels" which is a clone of "input_ids" with the appropriate maskings applied. https://github.com/microsoft/Olive/blob/ebe2f7ebbf8278698efddf45803e061fd097babc/olive/data/component/text_generation.py#L553

Reply:
When I print the value of the dataloader right after loading, I don't see the "labels" key.
```python
data_config = validate_config(config.calibration_data_config, DataConfig)
dataloader = data_config.to_data_container().create_dataloader(data_root)
print("dataset = ", next(iter(dataloader))[0])

_generates this output_
{'input_ids': tensor([[ 628,  796,  569,  ...,  416, 6343,   13]]), 'attention_mask': tensor([[1, 1, 1,  ..., 1, 1, 1]])}
```

Does the `DataConfig` description in config somehow drive the existence of this key in the generated output.

Reply:
Nevermind, the labels are keyed. It's the second member in the tuple.
```python
[{'input_ids': tensor([[ 628,  796,  569,  ...,  416, 6343,   13]]), 'attention_mask': tensor([[1, 1, 1,  ..., 1, 1, 1]])}, tensor([[ 628,  796,  569,  ...,  416, 6343,   13]])]
```